beacon-chain/state/stategen/migrate.go

```diff
@@ -5,152 +5,103 @@ import (
-			log.WithFields(logrus.Fields{
-				"slot":         stateSummary.Slot,
-				"archiveIndex": archivedPointIndex,
-				"root":         hex.EncodeToString(bytesutil.Trunc(r[:])),
-			}).Info("Saved archived point during state migration")
-		} else {
-			// Do not delete the current finalized state in case user wants to
-			// switch back to old state service, deleting the recent finalized state
-			// could cause issue switching back.
-			lastArchivedIndexRoot := s.beaconDB.LastArchivedIndexRoot(ctx)
-			if s.beaconDB.HasState(ctx, r) && r != lastArchivedIndexRoot && r != finalizedRoot {
-				if err := s.beaconDB.DeleteState(ctx, r); err != nil {
-					// For whatever reason if node is unable to delete a state due to
-					// state is finalized, it is more reasonable to continue than to exit.
-					log.Warnf("Unable to delete state during migration: %v", err)
-					continue
-				}
-				log.WithFields(logrus.Fields{
-					"slot": stateSummary.Slot,
-					"root": hex.EncodeToString(bytesutil.Trunc(r[:])),
-				}).Info("Deleted state during migration")
+			if err := s.beaconDB.SaveLastArchivedIndex(ctx, aIndex); err != nil {
+				return err
 			}
+			log.WithFields(
+				logrus.Fields{
+					"slot":          aState.Slot(),
+					"archivedIndex": aIndex,
+					"root":          hex.EncodeToString(bytesutil.Trunc(aRoot[:])),
+				}).Info("Saved state in DB")
 		}
 	}
 
-	// Update the split slot and root.
-	s.splitInfo = &splitSlotAndRoot{slot: finalizedSlot, root: finalizedRoot}
-	log.WithFields(logrus.Fields{
-		"slot": s.splitInfo.slot,
-		"root": hex.EncodeToString(bytesutil.Trunc(s.splitInfo.root[:])),
-	}).Info("Set hot and cold state split point")
-
-	return nil
-}
-
-// This recovers the last archived point. By passing in the current archived point, this recomputes
-// the state of last skipped archived point and save the missing state, archived point root, archived index to the DB.
-func (s *State) recoverArchivedPoint(ctx context.Context, currentArchivedPoint uint64) (uint64, error) {
-	missingIndex := currentArchivedPoint - 1
-	missingIndexSlot := missingIndex * s.slotsPerArchivedPoint
-	blks, err := s.beaconDB.HighestSlotBlocksBelow(ctx, missingIndexSlot)
-	if err != nil {
-		return 0, err
-	}
-	if len(blks) != 1 {
-		return 0, errUnknownBlock
-	}
-	missingRoot, err := ssz.HashTreeRoot(blks[0].Block)
-	if err != nil {
-		return 0, err
-	}
-	missingState, err := s.StateByRoot(ctx, missingRoot)
-	if err != nil {
-		return 0, err
+	// Migrate all state summary objects from state summary cache to DB.
+	if err := s.beaconDB.SaveStateSummaries(ctx, s.stateSummaryCache.GetAll()); err != nil {
+		return err
 	}
+	s.stateSummaryCache.Clear()
 
-	if err := s.beaconDB.SaveState(ctx, missingState, missingRoot); err != nil {
-		return 0, err
-	}
-	if err := s.beaconDB.SaveArchivedPointRoot(ctx, missingRoot, missingIndex); err != nil {
-		return 0, err
+	// Update finalized info in memory.
+	fInfo, e, err := s.epochBoundaryStateCache.getByRoot(fRoot)
```

Comment:
Please change `e` to `ok`.

```suggestion
	fInfo, ok, err := s.epochBoundaryStateCache.getByRoot(fRoot)
```

Reply:
addressed in #6488